### ISSUE TITLE
Suppress MatchingDeclarationName for file containing only an extension property

### ISSUE THREAD

**Issue by sunilson** [arturbosch/detekt#2187]
I have a .kt file that only has a single extension property like this

`val MyObject.myProperty = ...`

I want to call this file something like MyObjectExtensions.kt, but Detekt won't let me do that due to a `MatchingDeclarationName` error. Supressing doesn't seem to work though, not directly on the property or via `@file`. How could I suppress this error here?

**Comment by BraisGabin** [comment]
I just tried with this code and detekt doesn't show any warning: 
```kotlin
@file:Suppress("MatchingDeclarationName")

val Any.myProperty = ' '
```

Could you provide a project/file to reproduce your issue?

**Comment by jonas-arkulpa** [comment]
Found the issue: We have two files with the same name and the same issue in two different modules of our project. Suppressing was successful but then nearly the same error showed up for the other file and we kept on trying. 🤦‍♂
